### PR TITLE
ENG-2082: cleaning up warnings on jest

### PR DESCRIPTION
**Pull request by gReis89**
https://entando.myjetbrains.com/youtrack/issue/ENG-2082

All warnings being logged on tests were related to two issues:

```
React.createFactory() is deprecated and will be removed in a future major release. Consider using JSX or use React.createElement() directly instead.
```
This was being thrown by [Recompose](https://github.com/acdlite/recompose) a deprecated library that is no longer being supported, although Entando is not directly importing Recompose, but patternfly-v3 is, another deprecated library our system relies on which means we won't have an update that will solve this warning at point in time.

```
react-unsafe-component-lifecycles
```
This warning is basically React telling us that methods like `componentWill*` will be deprecated at some point in time, and we should get rid of it by refactoring code or changing to the `UNSAFE_` version.

So I wrote a mocking function that filters out these warnings to have a cleaner console

### CODE REVIEW

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **greis89**

If they are not already assigned, you can assign the PR to them by writing `/assign @greis89` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando%2Fapp-builder).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/entando/app-builder/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by gidesan** [comment]
This warning sound legit to me, because we're using and old version of `patternfly` with an old version of `recompose`. I'd avoid having errors in the future we're not able to easily find because of this workaround.

**Comment by gReis89** [comment]
This error is very noisy, it get logged on almost every test, it is not fixable because we can't move out from patternfly v3 without a full refactor on all our components, we can remove this filter once we start moving away from patternfly v3, but for know, I think we can knowledge and mute this warning for clearness purpose.

**Review by gidesan — APPROVED** [review]

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/entando/app-builder/pull/1060#pullrequestreview-645811721" title="Approved">gidesan</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando%2Fapp-builder).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/entando/app-builder/blob/master/OWNERS)~~ [gidesan]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
